internal/cloud/realm/realm.go

```diff
@@ -212,6 +254,142 @@ func isValidLocation(l Location) bool {
+
+	// GCP provider regions that we currently support
+	GCPProviderRegionUSCentral1  ProviderRegion = "gcp-us-central1"
+	GCPProviderRegionUSWest1     ProviderRegion = "gcp-us-west1"
+	GCPProviderRegionEuropeWest1 ProviderRegion = "gcp-europe-west1"
+	GCPProviderRegionAsiaSouth1  ProviderRegion = "gcp-asia-south1"
+)
+
+// ProviderRegionToLocation provides a quick lookup of a provider and region combination to
+// location values. This is necessary as the backing clusters no longer match our current
+// operating regions. In order to do backing cluster storage, we must leverage a lookup here.
+var ProviderRegionToLocation = map[ProviderRegion]Location{
+	ProviderRegionEmpty:              LocationVirginia,
+	AWSProviderRegionUSEast1:         LocationVirginia,
+	AWSProviderRegionUSWest2:         LocationOregon,
+	AWSProviderRegionEUCentral1:      LocationFrankfurt,
+	AWSProviderRegionEUWest1:         LocationIreland,
+	AWSProviderRegionAPSoutheast1:    LocationSingapore,
+	AWSProviderRegionAPSoutheast2:    LocationSydney,
+	AWSProviderRegionAPSouth1:        LocationMumbai,
+	AWSProviderRegionUSEast2:         LocationVirginia,
+	AWSProviderRegionEUWest2:         LocationIreland,
+	AWSProviderRegionSAEast1:         LocationSaoPaolo,
+	AzureProviderRegionEastUS2:       LocationVirginia,
+	AzureProviderRegionWestUS:        LocationOregon,
+	AzureProviderRegionWestEurope:    LocationFrankfurt,
+	AzureProviderRegionEastAsia:      LocationMumbai,
+	AzureProviderRegionSouthEastAsia: LocationSingapore,
+	GCPProviderRegionUSCentral1:      LocationVirginia,
+	GCPProviderRegionUSWest1:         LocationOregon,
+	GCPProviderRegionEuropeWest1:     LocationFrankfurt,
+	GCPProviderRegionAsiaSouth1:      LocationMumbai,
+}
+
+// ProviderRegion is the Realm app provider region
+type ProviderRegion string
+
+// String returns the ProviderRegion display
+func (p ProviderRegion) String() string { return string(p) }
+
+func (p ProviderRegion) Label() string {
+	return p.String()[strings.IndexByte(p.String(), '-')+1:]
+}
+
+// Type returns the ProviderRegion type
+func (p ProviderRegion) Type() string { return flags.TypeString }
+
+// Set validates and sets the ProviderRegion value
+func (p *ProviderRegion) Set(val string) error {
+	newProviderRegion := ProviderRegion(strings.ToLower(val))
+
+	if !isValidProviderRegion(newProviderRegion) {
+		return errInvalidProviderRegion
+	}
+
+	*p = newProviderRegion
+	return nil
+}
+
+// WriteAnswer validates and sets the ProviderRegion value
+func (p *ProviderRegion) WriteAnswer(name string, value interface{}) error {
+	var newProviderRegion ProviderRegion
+
+	switch v := value.(type) {
+	case core.OptionAnswer:
+		newProviderRegion = ProviderRegion(v.Value)
+	}
+
+	if !isValidProviderRegion(newProviderRegion) {
+		return errInvalidProviderRegion
+	}
+	*p = newProviderRegion
+	return nil
+}
+
+func isValidProviderRegion(p ProviderRegion) bool {
+	switch p {
+	case
+		ProviderRegionEmpty,
```

Comment:
[q] we allow this because it's the same as using the default through the api

Reply:
Yeah, also for creating a 'minimal' command, if it isn't required, then I think the cli should support it like it does for location. No strong feelings either way